core/dbt/config/utils.py

```diff
@@ -21,3 +27,47 @@ def parse_cli_vars(var_string: str) -> Dict[str, Any]:
     except ValidationException:
         fire_event(InvalidVarsYAML())
         raise
+
+
+def get_project_config(
+    project_path: str,
+    profile_name: str,
+    args: Namespace = Namespace(),
+    cli_vars: Optional[Dict[str, Any]] = None,
+    profile: Optional[Profile] = None,
+    user_config: Optional[UserConfig] = None,
+    return_dict: Boolean = True,
+) -> Union[Project, Dict]:
+    """Returns a project config (dict or object) from a given project path and profile name.
+
+    Args:
+        project_path: Path to project
+        profile_name: Name of profile
+        args: An argparse.Namespace that represents what would have been passed in on the
+            command line (optional)
+        cli_vars: A dict of any vars that would have been passed in on the command line (optional)
+            (see parse_cli_vars above for formatting details)
+        profile: A dbt.config.profile.Profile object (optional)
+        user_config: A dbt.contracts.project.UserConfig object (optional)
+        return_dict: Return a dict if true, return the full dbt.config.project.Project object if false
+
+    Returns:
+        A full project config
+
+    """
+    # Generate a profile if not provided
+    if profile is None:
+        # Generate user_config if not provided
+        if user_config is None:
+            user_config = read_user_config(flags.PROFILES_DIR)
+        # Update flags
+        flags.set_from_args(args, user_config)
+        profile = Profile.render_from_args(args, ProfileRenderer(cli_vars), profile_name)
```

Comment:
Looks like cli_vars is only used when a profile is not provided, is it safe to assume that the provided profile will always have cli_vars set?

Reply:
> is it safe to assume that the provided profile will always have cli_vars set
Nope, but the profile is valid wether or not it has `cli_vars` so we don't need to do anything special here.